### PR TITLE
feat(frontend): added admin schedule page

### PR DESCRIPTION
**Pull request by bravichandra12**
## Overview

1. This PR fixes or fixes part of [#90].
2. This PR does the following:
-  Adds Admin Schedule Page

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct
Screenshot (109)
Screenshot (110)

### CODE REVIEW

**Comment by Pratik2026** [comment]
@bravichandra12, I might not have explicitly mentioned but the issue is to create a complete schedule page by which I meant to create the cards component involved in this schedule page as well.

**Review by Pratik2026 — CHANGES_REQUESTED** [review]

**Comment by Pratik2026** [comment]
Add the cards component as well with dummy data along with the requested changes as mentioned above and make sure to keep the page responsive as currently it does not seem so.

File: frontend/src/store/useStore.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {create} from 'zustand';
+
+const useStore = create(set => ({
+  active: 0,
+  setActive: value => set({active: value}),
+}));
+
+export default useStore;
```
[the remark this answers is not in the document]
Comment (bravichandra12) [reply frontend/src/store/useStore.tsx:8]: Made the changes as mentioned

File: frontend/src/store/useStore.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import {create} from 'zustand';
+
+const useStore = create(set => ({
+  active: 0,
+  setActive: value => set({active: value}),
```
[the remark this answers is not in the document]
Comment (bravichandra12) [reply frontend/src/store/useStore.tsx:5]: Made the changes as mentioned

**Comment by Pratik2026** [comment]
@bravichandra12, The card component should have only one card so that we can use that card component as a reusable UI card.

**Comment by Pratik2026** [comment]
Also, currently, if we directly visit the schedule page then the selected tab is still the dashboard. This can be fixed by using `useEffect` in schedule page to change the `activeTab` state value according to the URL of the page.

**Comment by bravichandra12** [comment]
Ok I will do the changes as mentioned

**Comment by bravichandra12** [comment]
@Pratik2026 
The changes are made accordingly

**Comment by Pratik2026** [comment]
@bravichandra12, The changes look good to me. However, the layout is not responsive in mobile view. Currently, it's overflowing whenever you switch between the bus details and conductor details tab.

File: frontend/src/components/DetailsCards/BusCards/index.tsx [hunk 3]
```diff
@@ -58,7 +56,7 @@ const BusCards = () => {
     letter-spacing: 1.6px;
   `;
   const Card = styled(Paper)`
-    width: 435px;
+    max-width: 435px;
     height: 279px;
```
Comment (Pratik2026) [inline frontend/src/components/DetailsCards/BusCards/index.tsx:60]: remove this height.

File: frontend/src/components/DetailsCards/BusCards/index.tsx [hunk 4]
```diff
@@ -68,15 +66,15 @@ const BusCards = () => {
...
+    // height: 80px;
     background-color: #fbbc05;
     border-radius: 15px;
     padding-left: 0px;
     display: flex;
   `;
   const BusTag = styled(Box)`
-    width: 117px;
+    max-width: 117px;
     height: 20px;
```
Comment (Pratik2026) [inline frontend/src/components/DetailsCards/BusCards/index.tsx:78]: remove this height as well.

File: frontend/src/components/DetailsCards/BusCards/index.tsx [hunk 5]
```diff
@@ -68,15 +66,15 @@ const BusCards = () => {
     padding: 20px;
   `;
   const HeadBar = styled(Box)`
-    width: 395px;
-    height: 80px;
+    max-width: 395px;
+    // height: 80px;
```
Comment (Pratik2026) [inline frontend/src/components/DetailsCards/BusCards/index.tsx:70]: don't comment out, just completely remove all the commented one.

File: frontend/src/pages/AdminSchedule/index.tsx [hunk 6]
```diff
@@ -134,7 +134,7 @@ const AdminSchedule = () => {
                     xs: '1fr',
                     sm: '1fr',
                     md: '1fr',
-                    lg: '1fr 1fr ',
+                    lg: '1fr',
```
Comment (Pratik2026) [inline frontend/src/pages/AdminSchedule/index.tsx:137]: let's remove xl and set lg: '1fr 1fr'

**Review by Pratik2026 — CHANGES_REQUESTED** [review]

File: frontend/src/pages/AdminSchedule/index.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+              width="100%"
+              overflow="auto"
+              sx={{
+                mt: '1rem',
+              }}
+            >
+              <Box
+                sx={{
+                  width: '100%',
+                  height: '50rem',
```
Comment (Pratik2026) [inline frontend/src/pages/AdminSchedule/index.tsx:131]: remove this height property.

File: frontend/src/pages/AdminSchedule/index.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+                  pl: '1rem',
+                }}
+              >
+                <BusCards />
+                <BusCards />
+                <BusCards />
+                <Box
+                  sx={{
+                    width: '435px',
+                    height: '279px',
```
Comment (Pratik2026) [inline frontend/src/pages/AdminSchedule/index.tsx:150]: remove this height as well.

File: frontend/src/pages/AdminSchedule/index.tsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+              width="100%"
+              overflow="auto"
+              sx={{
+                mt: '1rem',
+              }}
+            >
+              <Box
+                sx={{
+                  width: '100%',
+                  height: '50rem',
```
Comment (Pratik2026) [inline frontend/src/pages/AdminSchedule/index.tsx:181]: remove this height

File: frontend/src/pages/AdminSchedule/index.tsx [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+                  gap: '40px',
+                  pl: '1rem',
+                }}
+              >
+                <ConductorCards />
+                <ConductorCards />
+                <Box
+                  sx={{
+                    width: '260px',
+                    height: '320px',
```
Comment (Pratik2026) [inline frontend/src/pages/AdminSchedule/index.tsx:198]: replace width with max-width and remove the height

**Comment by bravichandra12** [comment]
Made the changes as mentioned

**Comment by Pratik2026** [comment]
LGTM.

**Review by Pratik2026 — APPROVED** [review]

**Comment by Pratik2026** [comment]
merged, thanks for the contribution @bravichandra12

**Pull request merged**